server/image_response.js

```diff
@@ -22,19 +22,40 @@ module.exports = function(db, image_dir) {
                 // image found
                 let batchPath = path.join(image_dir, image.batches[0].toString());
                 let imagePath;
-                if (large) { imagePath = path.join(batchPath, image.hash + image.extension) }
+                // helper functions
+                let checker = (imgPath, errorFunction) => {
+                    fs.access(imgPath, fs.constants.F_OK, (err) => {
+                        if (err) {
+                            errorFunction(err);
+                        }
+                        else {
+                            res.sendFile(imgPath);
+                        }
+                    });
+                };
+                let noImage = (err) => {
+                    res.status(404).send({client: `Could not access image ${image.hash} in file system`,
```

Comment:
Nitpick: Can we change it to "Could not *find* image..." to fit with the 404 error better?